android/src/main/java/com/swmansion/reanimated/NativeMethodsHelper.java

```diff
@@ -57,9 +57,22 @@ public static void scrollTo(View view, double argX, double argY, boolean animate
 
     if (animated) {
       if (horizontal) {
-        ((ReactHorizontalScrollView) view).smoothScrollTo((int) x, (int) y);
+        View finalView = view;
+        view.post(new Runnable() {
+          @Override
+          public void run() {
+            ((ReactHorizontalScrollView) finalView).smoothScrollTo(x, y);
```

Comment:
The `finalView` should be `ReactHorizontalScrollView`, then we will get rid off the casting here.